activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -69,7 +69,11 @@ def select_rows(sql, name = nil, binds = [])
       end
       undef_method :select_rows
 
-      # Executes the SQL statement in the context of this connection.
+      # Executes the SQL statement in the context of this connection and returns
+      # the raw result from the connection adapter.
+      # Note: depending on your database connector, the result returned by this
+      # method may be manually memory managed. Consider using the exec_query
```

Comment:
The PG::Result object, for example, contains manually malloc'd memory that the Ruby GC doesn't see. This means that there is no GC pressure caused by this memory usage even though the VM's memory usage has increased.

The only way to free this memory is to call `clear` on the result, or manually trigger the GC (the result object has a GC hook that calls `clear`).